### PR TITLE
HOTT-1048: Add find commodity page

### PR DESCRIPTION
**Pull request by jebw**
### Jira link

[HOTT-1048](https://transformuk.atlassian.net/browse/HOTT-1048)

### What?

I have added/removed/altered:

- [x] Added a find commodity page
- [x] Redirect the existing sections page to the find commodity page when the feature flag is enabled
- [x] Updated various specs to test with both the feature flag off and on

### Why?

I am doing this because:

- We want to provide separate Find and Browse pages whilst keeping all existing links working

### Notes

Switch service banner is currently missing because that is being worked on separately and this is hidden behind a feature flag

### Screenshot

Screenshot from 2021-10-28 16-25-20

### CODE REVIEW

File: app/controllers/commodities_controller.rb [hunk 1]
```diff
@@ -4,6 +4,10 @@ class CommoditiesController < GoodsNomenclaturesController
   before_action :fetch_commodities, only: %i[show]
   before_action :set_session, only: %i[show]
 
+  def index
```
Comment (willfish) [inline app/controllers/commodities_controller.rb]: I'm not sure about this as the action for the find commodity page :thinking:

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -41,7 +41,8 @@ def govuk_header_navigation_item(active_class: '', &block)
   end
 
   def search_active_class
-    'active' if controller_name == 'search' || (controller_name == 'sections' && action_name == 'index')
+    'active' if action_name == 'search' ||
```
Comment (willfish) [inline app/helpers/application_helper.rb:44]: This feels a bit weird/horrible. I know it's what we're inheriting

File: app/webpacker/src/stylesheets/_feature-panel.scss [hunk 3]
```diff
@@ -9,4 +9,8 @@ $feature-panel-border-colour: #333;
   &.last-feature-panel {
     border: 0;
   }
+
+  &--shaded {
+    background-color: #f0f4f5;
```
Comment (willfish) [inline app/webpacker/src/stylesheets/_feature-panel.scss]: Should we explicitly name this color?

File: app/webpacker/src/javascripts/commodities.js [hunk 4]
```diff
@@ -798,7 +798,10 @@
                         data: obj
                       }
                     }]);
-                    $('.js-commodity-picker-select').parents('form:first').trigger("submit");
+
```
Comment (willfish) [inline app/webpacker/src/javascripts/commodities.js:801]: Do we want to write a test for this change?

File: app/helpers/application_helper.rb (hunk 2 above)
Comment (willfish) [reply app/helpers/application_helper.rb:44]: Chatted in slack about this. We've got quite a few dependency issues around layouts and controller action options that we can tidy up separately

File: app/webpacker/src/javascripts/commodities.js (hunk 4 above)
Comment (jebw) [reply app/webpacker/src/javascripts/commodities.js:801]: In principle yes, but the existing spec is skeletal at best. I'll need write a more complete spec for existing behaviour in order to add a spec for this small change. I can look at it this afternoon

File: app/webpacker/src/stylesheets/_feature-panel.scss (hunk 3 above)
Comment (jebw) [reply app/webpacker/src/stylesheets/_feature-panel.scss]: tbh - we shouldn't be using a colour outside the official palette - it adds nothing for accessibility. Following the designs though

**Review by willfish — APPROVED** [review]

**Pull request merged**